File: frontend/src/types/agent.ts
export type AgentRun = {
  id: number;
  project_id: number;
  run_type: string;
  status: string;
  steps_json: Record<string, unknown>;
  error_message: string | null;
  created_at: string;
  finished_at: string | null;
};

File: backend/app/schemas/agent.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class AgentRunRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    project_id: int
    run_type: str
    status: str
    steps_json: dict[str, Any]
    error_message: str | None
    created_at: datetime
    finished_at: datetime | None
